packages/merge-styles/src/IStyleSet.ts

```diff
@@ -6,6 +6,8 @@ export type Diff<T extends keyof any, U extends keyof any> = ({ [P in T]: P } &
 
 export type Omit<U, K extends keyof U> = Pick<U, Diff<keyof U, K>>;
 
+export type TrimToFunction<T> = T extends (...args: any[]) => any ? T : never;
```

Comment:
I'm a little puzzled by the `never`.

If someone were to define an interface `IStyles` as follows:
```ts
interface IStyles extends IStyleSet<IStyles> {
  root: IStyle;
  subComponentStyles: {
    choiceGroup: Partial<IChoiceGroupStyles>
  };
}
```
This technically is a valid IStyleSet as it represents the "object" portion.

But the conditional would then expectedly reduce it to a never... (e.g. below for classNames.subcomponentStyles.choicegroup)

```ts

const getStyles = (): Partial<IStyles> => ({
  root: {
    background: "red"
  },
  subComponentStyles: {
    choiceGroup: {
      root: {
        background: "green"
      }
    }
  }
});

const getClassNames = classNamesFunction<{}, IStyles>();

class App extends React.Component {
  public render() {
    const classNames = getClassNames(getStyles);
    return (
      <div className={classNames.root}>
        <ChoiceGroup
          options={[{ key: "1", text: "1" }, { key: "2", text: "2" }]}
          styles={classNames.subComponentStyles.choiceGroup}
        />
      </div>
    );
  }
}
```

Amazingly, there are no TS errors. The compiler accepted `never` to be assigned to styles, and the styles prop now has type `never`.

Reply:
The right thing is probably
`export type InferredStyleFunction<T> = T extends (...args: any[]) => any ? T : () => T; `